gitpr/gitpr.go

```diff
@@ -180,23 +182,38 @@ func (r Remote) GetOwnerSlashRepo() string {
 	return strings.Join(r.GetOwnerRepo(), "/")
 }
 
-// GetUsername queries GitHub for the username associated with a PAT.
-func GetUsername(pat string) string {
-	request, err := http.NewRequest("GET", "https://api.github.com/user", nil)
+// GetUsernameOrAppName returns the username for PAT auth (from /user)
+// or the app name for App-based auth (from /app), based on the useApp flag.
+func GetUsernameOrAppName(auther gitcmd.URLAuther, useApp bool) string {
```

Comment:
Why not just two functions? Or, it seems like this func should be able to figure it out based on what kind of `gitcmd.URLAuther` we get. Or maybe there can be a helper based on the flags that can do the right thing.

Reply:
agreed, it's pretty awkward. I'll split them up